lib/fog/vsphere/requests/compute/create_vm.rb

```diff
@@ -26,7 +26,12 @@ def create_vm attributes = { }
                             get_raw_cluster(attributes[:cluster], attributes[:datacenter]).resourcePool
                           end
           vmFolder      = get_raw_vmfolder(attributes[:path], attributes[:datacenter])
-          vm            = vmFolder.CreateVM_Task(:config => vm_cfg, :pool => resource_pool).wait_for_completion
+          # if any volume has a storage_pod set, we deploy the vm on a storage pod instead of the defined datastores
+          if pool = get_storage_pod(attributes)
```

Comment:
Could you do the assign and then check if the variable is `nil` in the conditional?